lib/configuration.rb

```diff
@@ -0,0 +1,20 @@
+class HL7::Configuration
+  attr_reader :segments
+
+  def initialize
+    @segments = Segments.new
+  end
+
+  class Segments
+    def add(segment_name, &block)
+      segment_class = Class.new(HL7::Message::Segment, &block)
+      HL7::Message::Segment.const_set(segment_name, segment_class)
+    end
+
+    def update(segment_name, &block)
+      HL7::Message::Segment
+        .const_get(segment_name)
+        .instance_eval(&block)
```

Comment:
I know we already have a bunch of `instance_eval`s in our code but I wish we didn't add more, specially next to a  `const_get` which in combination are a security risk because we are basically evaluating code from a user input.